packages/react-composites/src/composites/CallWithChatComposite/CallWithChatComposite.tsx

```diff
@@ -577,28 +581,52 @@ const CallWithChatScreen = (props: CallWithChatScreenProps): JSX.Element => {
     ]
   );
 
-  const onRenderChatContent = useCallback(
-    (): JSX.Element => (
+  const onRenderChatContent = useCallback((): JSX.Element => {
+    /* @conditional-compile-remove(breakout-rooms) */
+    return callWithChatAdapter.getState().chat?.threadId ? (
```

Comment:
instead of `getState`, use the onStateChange and store threadId in a useState. Reason being is this should re-render when the threadId changes and hence threadId should be in the dependency array of this callback

Reply:
I also think getState isn't reliable for to get timely re-renders. So I tried to store the threadId in a useState like you said:
![image](https://github.com/user-attachments/assets/2130a43b-b6c8-4c68-a6de-b80859c37ac8)

But for some reason it did not set the threadId to undefined (no spinner) like I expected as seen in this test run:
https://github.com/user-attachments/assets/c980f330-878e-4dfc-9601-45882fb061ea.

I will try other ways to not use `getState`

Reply:
Hmmm, your screenshot should've worked...

- Is callwithchatadpater.onStateChange not firing when the chat changes?
- Was threadId being set but for some reason the onRenderChatContent wasn't firing?

Reply:
Ah ok. I had to first unsub to main meeting chat adapter state changes to get it to work
![image](https://github.com/user-attachments/assets/c3fcde0c-bf83-44fe-8a0c-499e657a3739)
The main meeting chat adapter would just update the callWithChatAdapterState right after I unset the chat state.
Also I just used the existing useState `isChatInitialized` to decide whether to show the spinner.
Looks much better now. Thanks!